pkg/platform/telemetry/telemetry.go

```diff
@@ -0,0 +1,187 @@
+package telemetry
+
+import (
+	"bytes"
+	"net/http"
+	"runtime"
+	"time"
+
+	"go.keploy.io/server/pkg/models"
+	"go.keploy.io/server/utils"
+	"go.uber.org/zap"
+)
+
+var teleUrl = "https://telemetry.keploy.io/analytics"
+
+type Telemetry struct {
+	Enabled        bool
+	OffMode        bool
+	logger         *zap.Logger
+	InstallationID string
+	store          FS
+	KeployVersion  string
+	GlobalMap      map[string]interface{}
+	client         *http.Client
+}
+
+func NewTelemetry(enabled, offMode bool, store FS, logger *zap.Logger, KeployVersion string, GlobalMap map[string]interface{}) *Telemetry {
+
+	tele := Telemetry{
+		Enabled:       enabled,
+		OffMode:       offMode,
+		logger:        logger,
+		store:         store,
+		KeployVersion: KeployVersion,
+		GlobalMap:     GlobalMap,
+		client:        &http.Client{Timeout: 10 * time.Second},
+	}
+	return &tele
+}
+
+func (tel *Telemetry) Ping(isTestMode bool) {
+	if !tel.Enabled {
+		return
+	}
+
+	go func() {
+		defer utils.HandlePanic()
+		for {
+			var count int64
+			var id string
+
+			if tel.Enabled {
+				id, _ = tel.store.Get(true)
+				count = int64(len(id))
+			}
+			event := models.TeleEvent{
+				EventType: "Ping",
+				CreatedAt: time.Now().Unix(),
+			}
+			tel.InstallationID = id
+
+			if count == 0 {
+				//Check in the old keploy folder.
+				id, _ = tel.store.Get(false)
+				count = int64(len(id))
+				if count == 0 {
+					bin, err := marshalEvent(event, tel.logger)
+					if err != nil {
+						break
+					}
+					resp, err := http.Post(teleUrl, "application/json", bytes.NewBuffer(bin))
+					if err != nil {
+						tel.logger.Debug("failed to send request for analytics", zap.Error(err))
+						break
+					}
+					installation_id, err := unmarshalResp(resp, tel.logger)
+					if err != nil {
+						break
+					}
+					id = installation_id
+					tel.InstallationID = id
```

Comment:
For the case where you get new installation_id from telemetry server you are setting the id twice . This is wrong . This was the 3rd time I am commenting on the same line. Please refer the code from v1 branch from which you took this function.

Reply:
updated